### PR TITLE
new concepts page

### PR DESCRIPTION
**Pull request by evilnick**
could do with some pretty graphics at some point!

### CODE REVIEW

**Comment by evilnick** [comment]
@pmatulis @matthewhelmke

File: src/en/juju-concepts.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+
+!!! Note: This page is a work in progress, as part of the work towards 
+documenting Juju 2.0.
+
+# Juju concepts
+
+## The Juju client
+
+The Juju package which you install and run from your local computer is sometimes
```
Comment (pmatulis) [inline src/en/juju-concepts.md]: Suggestion:

The Juju client is CLI software that is used to manage Juju, whether as an administrator or as a regular user. It is installed, via an APT package (juju2), onto your personal workstation.

Consider mentioning Juju GUI?

File: src/en/juju-concepts.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+## Clouds
+
+In Juju terms, the cloud is the environment which provides resources to be used
+by Juju - machines and storage. This includes traditional public clouds, such
+as Amazon Web Services, Google Compute Engine and Microsoft Azure, as well as
+(possibly private) clouds such as OpenStack.
+
+It also includes things which may not strictly speaking be clouds, but Juju can
+treat as a cloud, such as [MAAS][maas] and your local computer using LXD.
```
Comment (pmatulis) [inline src/en/juju-concepts.md]: and [LXD][link] (local computer).

File: src/en/juju-concepts.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+
+## The Juju controller
+
+The controller is a special instance which Juju will create in a cloud when you
```
Comment (pmatulis) [inline src/en/juju-concepts.md]: s/instance/machine

File: src/en/juju-concepts.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+run the bootstrap command. The controller is effectively a Juju agent which runs
```
Comment (pmatulis) [inline src/en/juju-concepts.md]: this is not the place to mention commands

File: src/en/juju-concepts.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+A model is the space within which you can deploy services. A controller can 
+manage several models, and the models themselves can even be shared amongst
+multiple users.
+
+## Charms
+
+The magic behind Juju is a collection of software components called Charms. They
+contain all the instructions necessary for deploying and configuring 
+cloud-based services. The charms publicly available in the online Charm Store
```
Comment (pmatulis) [inline src/en/juju-concepts.md]: provide link to store

File: src/en/juju-concepts.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+A model is the space within which you can deploy services. A controller can 
+manage several models, and the models themselves can even be shared amongst
+multiple users.
+
+## Charms
+
+The magic behind Juju is a collection of software components called Charms. They
+contain all the instructions necessary for deploying and configuring 
+cloud-based services. The charms publicly available in the online Charm Store 
+represent the distilled DevOps knowledge of experts. Charms make it easy to
```
Comment (pmatulis) [inline src/en/juju-concepts.md]: I dislike this slogan.

DevOps is a complex process, describing how people in various teams (including end-users) should interact with one another for the benefit of providing the best software and user experience possible.

File: src/en/juju-concepts.md [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+knowledge of how to connect services together using common interfaces they
+share. For example, WordPress supports an 'http' interface (for serving the 
+website) and a 'db' interface (for the database which stores the content of
+the site). Any other service which can support these interfaces(e.g haproxy for
+the website and MySQL for the database) can connect to the WordPress charm in 
+a meaningful way.
+
+## Bundles
+
+A bundle is a ready-to-run collection of services that have been modelled to
```
Comment (pmatulis) [inline src/en/juju-concepts.md]: might want to avoid alluding to model here

File: src/en/juju-concepts.md [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+reliably, and of course you can still use Juju to manage them after deployment 
+to scale them any way you wish.
+
+Juju can deploy bundles directly from the command line or the Juju GUI can both
+deploy and save bundles. See the [documentation on bundles][bundles] for more
+information.
+
+## Machines
+
+A machine is the term used to describe a running instance in the cloud. Machines
```
Comment (pmatulis) [inline src/en/juju-concepts.md]: ok, may want to ignore my earlier machine comment. so an LXD container is considered an instance in an LXD cloud? we're really stretching words here. i'm not sure if this helps or hinders

File: src/en/juju-concepts.md [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+will usually house a single unit of a deployed service, but this is not always 
+the case. If directed by the user a machine may house several units (e.g. to 
+conserve resources) or possibly no units at all: a machine can be created 
+independently of services, though usually this is with the intention of 
+eventually running a service on it!
+
+## Units
```
Comment (pmatulis) [inline src/en/juju-concepts.md]: Add "and services" to header?

**Comment by pmatulis** [comment]
Review complete

File: src/en/juju-concepts.md (hunk 8 above)
Comment (evilnick) [inline src/en/juju-concepts.md]: in an LXD cloud, yes - it is a 'machine'. It appears as a machine in all the Juju command output.  I understand your desire for exactitude, but until all the juju commands output 'machines' with a little asterisk next to them, pointing out that they are in reality containers in the case of an LXD cloud, I think we have to not include caveats in the definitions which try to help describe those operations.

File: src/en/juju-concepts.md (hunk 6 above)
Comment (evilnick) [inline src/en/juju-concepts.md]: objection noted.

File: src/en/juju-concepts.md (hunk 7 above)
Comment (evilnick) [inline src/en/juju-concepts.md]: I thought about it, but that's exactly what they are - like a snapshot of a model

**Comment by evilnick** [comment]
I think I changed all the things you wanted which it is within my remit to change

**Pull request merged**